### DISCUSSION TITLE
Typing `onError` for `defaultOptions`

### DISCUSSION THREAD

**Discussion by sethmbrown** [TanStack/query#4233] [Q&A]
I'm trying to type the `error` object in the `defaultOptions` of my `queryClient`

Right now I have `const defaultOptions = DefaultOptions<AxiosError>` . When passing `deafultOptions` into `new QueryClient`, I get the following TS error - 

```
Type 'DefaultOptions<AxiosError<unknown, any>>' is not assignable to type 'DefaultOptions<unknown>'. 
```

What is the correct way to handle typing the error object globally here? Thanks!

(Tried to post this on the community discord, but it did not appear to work)

**Comment by TkDodo** [comment]
I don't think you can type the defaultOptions. A runtime check with `instanceof Error` is safer and easier to achieve

**Reply by sethmbrown** [reply]
Thank you for the response! 

Ok, so each `onError` will required an `instanceOf Error` check?

**Reply by TkDodo** [reply]
isn't there only one in the defaultOptions ?
